instructions.md

```diff
@@ -0,0 +1,203 @@
+- Set the new state with the setter function using the spread operator.
+```js
+[...oldStateArray, newArrayObject]
+```
+
+
+### Task #3 
+
+Based on the given requirements, implement `ContactsPage` as a stateful component to handle the logic for adding new contacts and listing current contacts.
+
+This component will:
+- Have two props: 
+  - The current list of contacts
+  - A function for adding a new contact
+- Render a `ContactForm` and a `TileList`.
+- Keep track of the current name, phone, and email entered into the form.
+- Check for duplicates whenever the name in the form changes and indicate the name is a duplicate.
+- Only add a new contact on form submission if it does not duplicate an existing contact's name.
+- A successful submission should clear the form.
+
+### Hint
+
+- Add `{ useState, useEffect }` to the `'react'` import statement
+- Import `ContactForm` and `TileList`
+- Make `props` a function parameter
+- The contact info state variables should default to an empty string using `useState('')`
+- The state variable to track a duplicate name should be set to false using `useState(false)`
+- Implement a call to `useEffect`, where the callback function sets the duplicate state variable to `true` if the name state variable is already in the contacts list
+- An `if` statement should be added to `handleSubmit` that checks the duplicate state variable is `false`. The body of the `if` statement should call the add contact function in `props` and set the contact info state variables to `''` 
+
+### Task #4 
+
+Based on the given requirements, implement `ContactForm` as a stateless component that renders a web form to collect the necessary contact information.
+
+This component will:
+- Have 7 props: 
+  - The variables for contact name, phone and email 
+  - The setters for contact name, phone and email 
+  - The function that handles the form submission
+- Render a `form` with:
+  - The `onSubmit` attribute set
+  - 3 controlled `input` components, one for each piece of contact data
+  - A submit button
+- Include a `pattern` attribute to the phone `input` with [a regex](https://regexlib.com/Search.aspx?k=phone&c=-1&m=-1&ps=20) that matches the phone locale of your preference.
+
+### Hint
+
+- Make `props` a function parameter
+- Use `<form onSubmit={props.handlerFunction} >` to create the html form
+- The children of the form should be `<input>`s of `type` `text`, `tel`, `email` and `submit`
+- The `value` attribute of the 3 controlled `input`s should be set to the associated variable in `props`
+- The `onChange` attributes of the 3 controlled `input`s should be set to a callback function that passes `e.target.value` to the the associated setter function
+- One example of a phone regex is the US locale: `"[1-9][0-9]{2}-[1-9][0-9]{2}-[0-9]{4}"`
+
+### Task #5 
+
+Based on the given requirements, implement `TileList` as a stateless component that renders a list of `Tile` components using an array of objects.
+
```

Comment:
Could you add in a sentence or two explaining that:
* TileList` is a generic component that will be shared by the contacts page and the appointments page. 
* Each Tile will represent a single contact or appointment